src/communication/context.hpp

```diff
@@ -95,16 +92,13 @@ class ServerContext final {
   ServerContext(ServerContext &&other) noexcept;
   ServerContext &operator=(ServerContext &&other) noexcept;
 
-  // Destructor that handles ownership of the SSL object.
-  ~ServerContext();
-
   SSL_CTX *context();
+  SSL_CTX *context_clone();
 
   bool use_ssl();
 
  private:
-  bool use_ssl_;
-  SSL_CTX *ctx_;
+  std::optional<SSL_CTX *> ctx_;
```

Comment:
I wanted to touch this part as minimally as possible because the focus of the PR is the WSS. 
The context is used by multiple servers so I would need to change all of the interactions with it.

Reply:
Then don't, it's not really worth it!